### PR TITLE
MNT: Remove unused imports

### PR DESCRIPTION
**Pull request by oscargus**
## PR Summary

Found some unused imports.

## PR Checklist

**Tests and Styling**
- [N/A] Has pytest style unit tests (and `pytest` passes).
- [N/A] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [N/A] New features are documented, with examples if plot related.
- [N/A] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [N/A] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
- [N/A] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

### CODE REVIEW

**Comment by QuLogic** [comment]
It looks like this is caused by [this blanket exception](https://github.com/matplotlib/matplotlib/blob/main/.flake8#L56); perhaps that can be removed now?

**Comment by oscargus** [comment]
> It looks like this is caused by [this blanket exception](https://github.com/matplotlib/matplotlib/blob/main/.flake8#L56); perhaps that can be removed now?

There are still unused imports in backends, but ones that I believe are required, like the FigureManagers etc. I guess one can mark the lines, which I am strongly in favour of instead of the file, but it will take some time...

**Comment by anntzer** [comment]
Perhaps you can remove F401 for all backends from `.flake8` (and put in a more targeted exclusion if needed)?

**Review by QuLogic — APPROVED** [review]

**Comment by oscargus** [comment]
New change where I started to use `__all__`, which seems like it hasn't been used earlier. For some files it is a bit of a hassle, so I selected the relatively easy ones. 

There is a minor risk for breakage. For example, it is in current main possible to import numpy as:
```
from matplotlib.tri import np
```
This is no longer possible after this change.

File: lib/mpl_toolkits/axisartist/axislines.py [hunk 1]
```diff
@@ -46,7 +46,7 @@
 import matplotlib.axes as maxes
 from matplotlib.path import Path
 from mpl_toolkits.axes_grid1 import mpl_axes
-from .axisline_style import AxislineStyle
+from .axisline_style import AxislineStyle  # noqa
```
Comment (oscargus) [inline lib/mpl_toolkits/axisartist/axislines.py:49]: Maybe one should deprecate this? Not sure if someone is actually importing it from here?

**Comment by oscargus** [comment]
Converting to draft until I have had time to think about the `tri` documentation issue.

**Review by QuLogic — APPROVED** [review]

File: lib/matplotlib/tri/__init__.py [hunk 2]
```diff
@@ -2,11 +2,22 @@
...
-from .triangulation import *
-from .tricontour import *
-from .tritools import *
-from .trifinder import *
-from .triinterpolate import *
-from .trirefine import *
-from .tripcolor import *
-from .triplot import *
+from .triangulation import Triangulation
+from .tricontour import TriContourSet,  tricontour, tricontourf
```
[the remark this answers is not in the document]
Comment (oscargus) [reply lib/matplotlib/tri/__init__.py:6]: Got it sorted locally, but waited with pushing. Will do.

**Review by anntzer — APPROVED** [review]

**Pull request merged**